src/virtual_elements.ts

```diff
@@ -129,12 +129,13 @@ function elementOpen(
     applyStatics(node, data, statics);
   }
 
-  if (arguments.length <= ATTRIBUTES_OFFSET && !data.hasAttrsArr()) {
+  const attrsLength = Math.max(0, arguments.length - ATTRIBUTES_OFFSET);
+  if (!attrsLength && !data.hasAttrsArr()) {
```

Comment:
Seems to make sense to do:

```javascript
const hadNoAttrs = data.hasEmptyAttrsArr();

if (!attrsLength && hadNoAttrs) {
  return node;
}

// use hadNoAttrs instead of isNew
```

If we had no attrs, and there are no new attrs, there is no need to diff, even if the attrs array is present.

For the other usage in the statics function, maybe we can just use `hasEmptyAttrsArr` instead of `hasAttrs` and just drop the function entirely.

Reply:
That's a great idea. And yes, we can drop `hasAttrs` entirely with this. Done.